workspaces/libnpmexec/lib/index.js

```diff
@@ -41,6 +41,10 @@ const missingFromTree = async ({ spec, tree, flatOptions, isNpxTree }) => {
     // registry spec that is not a specific tag.
     const nodesBySpec = tree.inventory.query('packageName', spec.name)
     for (const node of nodesBySpec) {
+      // continue if node is not at specified depth, no need to check further
+      if (depth !== undefined && node.depth !== depth) {
```

Comment:
This is a great solution.  However we are currently not really using `depth` so much as telling this function to only look at depth 0.

It would probably make more sense in the future when reading this if this was called something like `shallow` and was a Boolean.  and then `if (shallow && node.depth)` would be the test for continuing.

Reply:
Makes sense.. Corrected in latest push.